tests/unit/test-util/test.artifacts.js

```diff
@@ -55,4 +56,43 @@ describe('prepareArtifactsDir', () => {
     }
   ));
 
+  it('throws error when checking the directory if lacks writing permissions',
+    () => withTempDir(
+    (tmpDir) => {
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      return fs.mkdir(tmpPath, '0622').then(() => {
```

Comment:
@kumar303 to test this change we have two possible strategies (which are already used in the project):

- recreate the scenario on the real filesystem like in this test (in a temp dir that is automatically removed after the test)
- inject the dependency (`fs` in this case) and use sinon to fake all the error scenarios without recreating anything on disk

I would be ok (e.g. test everything using sinon) to use just one of the two approaches, because using both is probably going to be confusing.

The scenarios to be tested are:

- the artifact dir exists but it is not a directory
- the artifact dir exists but it is a file
- the artifact dir exists but we lack the permissions to write files in it
- the artifact dir do not exist, but we lack the permission to create it

which one of the two approach do you prefer?